lib/inputstreamhelper.py

```diff
@@ -298,7 +298,13 @@ def _chromeos_offset(self, bin_path):
 
             # Find the largest partition, and store the offset
             # NOTE: For parted we need to remove the 's' unit
-            partition_size = int((partition.get('Size') or partition.get('Blocks')).replace('s', ''))
+            try:
+                partition_size = int((partition.get('Size') or partition.get('Blocks')).replace('s', ''))
+            except ValueError:
```

Comment:
We need to understand why this is not ok. Otherwise this code will become unmaintainable.
User A will fix what works for him, user B will undo and fix to make it work for him.
That is a recipe for disaster.

Reply:
The output from fdisk is not what was expected:
```

GPT PMBR size mismatch (7155868 != 2981935) will be corrected by w(rite).
Disk /home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin: 1,4 GiB, 1526751232 bytes, 2981936 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: AF84116D-1C9E-324D-BE87-A662A9751CD6

Device                                                                                                                                          Start     End Sectors  Size Type
/home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin1  2961408 2981887   20480   10M Microsoft basic data
/home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin2    20480   53247   32768   16M ChromeOS kernel
/home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin3   319488 2961407 2641920  1,3G ChromeOS root fs
/home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin4    53248   86015   32768   16M ChromeOS kernel
/home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin5   315392  319487    4096    2M ChromeOS root fs
/home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin6    16448   16448       1  512B ChromeOS kernel
/home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin7    16449   16449       1  512B ChromeOS root fs
/home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin8    86016  118783   32768   16M Microsoft basic data
/home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin9    16450   16450       1  512B ChromeOS reserved
/home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin10   16451   16451       1  512B ChromeOS reserved
/home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin11      64   16447   16384    8M unknown
/home/kodi/.kodi/userdata/addon_data/script.module.inputstreamhelper/tmp/chromeos_12239.67.0_nyan-big_recovery_stable-channel_big-mp-v2.bin12  249856  315391   65536   32M EFI System

Partition table entries are not in disk order.
```

This is the whole output of the initial `fdisk -l`. You can confirm all the issue i tried to fix. I don't know about parted as it wasn't installed on the system. Only you (maintainer) can know the difference i guess?

Right now, the splitting for 2 or 1 spaces could be seperated if fdisk or parted is used?
As for the except part, it's because of the "Partition table entries...".

Everything is not an "user" error, just a new chrome os image for aarch64 that broke the parsing.

Reply:
We will have to parse differently for `fdisk` and `parted`. It would make the parted parsing a lot more simple. And all the complexity can go into trying to handle `fdisk`.

The current parsing was tested on LibreELEC, RHEL7 and Travis. Using fdisk and parted (whatever was available). So the only path forward is to make a library with output and unit tests, so that change don't break known output. And it should have had some more testing before being released. It was intended for v0.5.0.

Reply:
Maybe we can revert this PR?

Reply:
Reverting https://github.com/emilsvennesson/script.module.inputstreamhelper/commit/0993eef6f5660d871559428cf89560bc9bff8f29 ?

Reply:
Yup. It may then fail locally, but it works as before.